serving/docs/lmi/tutorials/trtllm_finding_max_num_tokens_tutorial.md

```diff
@@ -0,0 +1,125 @@
+| Mistral 7B    | g5.12xl  | 1                      | 3287                 |  
+| Mistral 7B    | g5.12xl  | 4                      | 410000               | 
+| Falcon 7B     | g5.12xl  | 1                      | 2835                 | 
+| LLAMA 2 13B   | p4d.24xl | 4                      | 476718               | 
+| LLAMA 2 70B   | p4d.24xl | 8                      | 208125               | 
+| Mixtral 8x7B  | p4d.24xl | 4                      | 113906               | 
+| Falcon 40B    | p4d.24xl | 4                      | 152343               | 
+| CodeLLaMA 34B | p4d.24xl | 4                      | 295781               | 
+| CodeLLaMA 34B | p4d.24xl | 8                      | 384375               | 
+
+## Find the number with LMI-TRTLLM container solution
+
+Before finding the precise number, you can simply try the following formula by yourself to estimate:
+
+```
+max_num_tokens = (avg_input + avg_output) * avg_concurrency
+```
+And then try this number with LMI solution and see if it works. 
+If not working, go with a binary search approach by reducing halves until you get the number you want.
+
+We also created a simple function to allow you do this with LMI container.
+
+### Step 1: Launch the container
+
+Using your team AWS account and login with DLC account:
+
+```
+aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 763104351884.dkr.ecr.us-east-1.amazonaws.com
+```
+
+Then simply do
+
+```
+docker run -it --runtime=nvidia --gpus all --shm-size 12gb \
+-p 8080:8080 \
+-v /opt/dlami/nvme/large_store:/opt/djl/large_store \
+-v /opt/dlami/nvme/tmp/.cache:/tmp/.cache \
+763104351884.dkr.ecr.us-east-1.amazonaws.com/djl-inference:0.27.0-tensorrtllm0.8.0-cu122 /bin/bash
+```
+
+Here we assume you are using g5, g6, p4d, p4de or p5 machine that has NVMe disk available. 
+Mounting to that disk will bring you several terabytes of spaces.
+
+### Step 2: Start running with `max_token_finder`
+
+The following step will be executed inside the container.
+
+You can pre-download your huggingface model to `/opt/djl/large_store` or simply fetch from huggingface.
+You can create a file named `finder.py`.
+
+```python
+from tensorrt_llm_toolkit.utils.utils import max_token_finder
+
+tensor_parallel_degree = 4
+model_id = "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+
+properties = {
+    "model_id": model_id,
+    "tensor_parallel_degree": tensor_parallel_degree,
+}
+
+
+model, tp, max_tokens = max_token_finder(properties)
+print(f"Summary:\nmodel: {model}\n tp: {tp}\n max_tokens: {max_tokens}")
+```
+
+You can override `model_id` to
+- local path like `/opt/djl/large_store/<my_downloaded_model_folder>`
+- HF model id like `TinyLlama/TinyLlama-1.1B-Chat-v1.0`
+
+`tensor_parallel_degree` means the number of gpu you want to use to parallel computation.
+The larger value you set, the bigger max_token the model will be able to support. Finally, just run with:
+
+```
+python3 finder.py
```

Comment:
Not for this PR, but have we considered including this as an optional step in the compilation process?

It would be great if we could trigger this automatically (if user requests, or we could enable by default) as part of the compilation process

Reply:
This takes some time to finish, so that's why we didn't automate this